### ISSUE TITLE
draw_center unchecked doesn't work with shaders

### ISSUE THREAD

**Issue by rainlizard** [godotengine/godot#66980]
### Godot version

v4.0.beta2.official [f8745f2f7]

### System information

Windows 10, default, GTX 970

### Issue description

I'm using a PanelContainer with a StyleBoxTexture. In 4.0 when I uncheck Draw Center, it doesn't affect the image and continues to draw the center. In 3.5 it works.

![shader-border 3 5](https://user-images.githubusercontent.com/15337628/194304745-111771fa-84a5-4243-9ac5-aadcfc4881db.png)
![shader-border 4 0](https://user-images.githubusercontent.com/15337628/194304758-e54a9fc3-5f42-4e62-958d-fdf86cae39a6.png)
(the TextureRect node is just the Godot icon, it's irrelevant)

### Steps to reproduce

shader doesn't have to be anything special
```
shader_type canvas_item;

void fragment() {
	COLOR = texture(TEXTURE, UV);
}
```

### Minimal reproduction project

[shader-border 3.5.zip](https://github.com/godotengine/godot/files/9724512/shader-border.3.5.zip)
[shader-border 4.0.zip](https://github.com/godotengine/godot/files/9724508/shader-border.4.0.zip)

**Comment by veryprofessionaldodo** [comment]
Interesting, as the preview correctly shows the result. I'll look into it, but I've never touched anything on this side, so I'll get exploring, and report back anything that I've noticed.

**Comment by rainlizard** [comment]
Hope you find something. `NinePatchRect` doesn't seem to be a workaround and is broken in the same way, so it looks like (fancy) borders in general are broken.

**Comment by veryprofessionaldodo** [comment]
Yeah, it's quite tricky. By the way, have you ever set up a debug configuration using VS Code for the Godot Engine? I don't mean to debug an app, I mean the entire engine itself. Spent a few hours yesterday trying to get that set up, because it's a non trivial issue, and a debugger would be super useful.

**Comment by rainlizard** [comment]
I found something interesting, if you multiply the final `COLOR` rather than set it, then it behaves more like it did back in 3.5

![Untitled](https://user-images.githubusercontent.com/15337628/194677526-ae6dc181-5776-42fb-8342-43d3aedbe540.png)


But now it's darker than it should be for some reason. I tried adjusting the grey background but it didn't affect it. I also tried `render_mode blend_add;` but then it's too bright, though it is somewhat usable like this at least since you can always adjust its brightness.

But I wouldn't consider this to be the solution, as it's not user-friendly to expect people to figure out that they suddenly need to multiply `COLOR` in their shaders. I think forcing the multiply to `COLOR` behind the scenes if draw_center is unchecked might be a good idea. If the darkness of the image can be resolved.

**Comment by Zireael07** [comment]
I know COLOR behavior was changed in 4.0, by design and this is likely a side effect of this
